src/phpDebug.ts

```diff
@@ -325,6 +330,30 @@ class PhpDebugSession extends vscode.DebugSession {
         }
         this.sendResponse(response)
     }
+    /**
+     * Opens the urls specified in configuration, either with start or stop query string
+     * @param {boolean} start
+     */
+    private _openUrl(start: boolean) {
+        if (!this._args.openUrl) {
+            return
+        }
+        const myurl = new url.URL(this._args.openUrl)
+        if (start) {
+            myurl.searchParams.append('XDEBUG_SESSION_START', 'vscode')
+        } else {
+            myurl.searchParams.append('XDEBUG_SESSION_STOP', 'vscode')
+        }
+
+        const proc = opn(myurl.toString())
+        proc.then(child => {
+            child.stderr.on('data', (chunk: string) => {
+                this.sendEvent(new vscode.OutputEvent('openUrl: ' + chunk + '\n'))
+            })
+        }).catch((reason: any) => {
+            this.sendEvent(new vscode.OutputEvent('openUrl-error: ' + util.inspect(reason) + '\n'))
+        })
```

Comment:
Instead of loggint the error here, it’s better to make this an async function and have the callers await it, so an error is reported as a response to the request

Reply:
Not sure what exactly you mean here... so i should remove the .catch and just return the result?
Also should the then be removed too?

Reply:
See my two suggestions above. In addition you'd remove the `.catch()` and "unwrap" the `.then()`.